urlCutover.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
```

Comment:
This repo is accessible to customers but I think that only Pathify employees should run this script as it currently stands (automatically applying changes).  To make it safer, what if it showed a diff of the changes it made after the `awk` command and then prompted the user with, "do you want to apply these changes to this Flow server?"

One interesting thing is that in order for this script to work, I'm pretty sure that `~/creds/customer.flow` would have to be updated.  I'm not sure if the value in that file could be used to validate the proposed URL but maybe it could...

I see this is a draft, so I'm probably getting ahead of myself, but what if the script also made sure that oldUrl and newUrl met some minimal requirements of resembling a URL.  Just to protect against: `urlCutover customer <non-url that appears in many places> <new-url>`. 

There is another repo for our own internal scripts that I can refer you to in another channel if you like.

Reply:
Hey @timjstewart1024! Somehow I missed this comment. Many thanks for the context and proposals.

I was not aware that this was a customer-facing repo (it's my first time encountering such an amenity, as well. I'm a fan of the convention). This explains your hesitation regarding an incomplete comment in the `batchUpload` script. I agree with your assessment, this sort of orchestration likely belongs in a private repository, so I'd be keen to see the internal scripts repository.

Thanks for the ideas about how we could put up guard rails to protect against a faulty URL change. Agreed with everything you've suggested! 

I'm tempted to move the logic into a Python script, as I think we can implement some more robust validation with a pip library like [Pydantic](https://docs.pydantic.dev/latest/). Further, I can foresee a situation in which this is a one- or zero-click procedure on the flow side, that is triggered externally, leverages the `gsutil` to watch changes to certificate validation, updates the community repo once that's complete, and (one can dream) posts validation to the cutover thread in Slack upon completion. It'll take some work to get there, but it's within reach. Open to any and all input on these musings!